nmrfx-structure/src/main/java/org/nmrfx/structure/chemistry/energy/Dihedral.java

```diff
@@ -425,45 +425,29 @@ public void setBoundaries(final double sigma, boolean useDegrees, double maxRang
             inputSigma[i] = sigma;
         }
         int aStart = nPseudoAngles * 2 * stepSize;
+        List<String> parentNamesToScale = Arrays.asList("P", "O5'", "C5'", "C4'", "O3'", "CA", "C");
         for (int iAtom = 0; iAtom < angleAtoms.size(); iAtom++) {
             Atom atom = angleAtoms.get(iAtom).daughterAtom;
             // temporarily set everything to use sigma, till we make sure counting is correct
             if (atom == null) {
-                System.out.println("daughter null " + angleAtoms.get(iAtom).daughterAtom.getFullName());
-            }
-            if (atom.parent == null) {
-                System.out.println("parent null " + angleAtoms.get(iAtom).daughterAtom.getFullName());
-            }
-            if (atom.parent.getName().equals("P") || atom.parent.getName().equals("O5'") || atom.parent.getName().equals("C5'") || atom.parent.getName().equals("C4'") || atom.parent.getName().equals("O3'")) {
-                inputSigma[aStart++] = sigma / backBoneScale;
-                if (sinCosMode) {
-                    inputSigma[aStart++] = sigma / backBoneScale;
-                }
-            } else if (atom.parent.getName().equals("CA") || atom.parent.getName().equals("C")) {
-                inputSigma[aStart++] = sigma / backBoneScale;
-                if (sinCosMode) {
-                    inputSigma[aStart++] = sigma / backBoneScale;
-                }
+                throw new IllegalStateException("daughter null " + angleAtoms.get(iAtom).daughterAtom.getFullName());
```

Comment:
@brucejohnson suggestions for message? This is what was here before but I think that this message will result in a NPE for the getFullName() call as well since Atom atom = angleAtoms.get(iAtom).daughterAtom; and atom is null,